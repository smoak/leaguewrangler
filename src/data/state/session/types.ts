export interface Session {
  readonly userToken: string;
}
