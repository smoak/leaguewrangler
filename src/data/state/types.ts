import { Session } from './session/types';

export * from './session/types';

export interface ApplicationState {
  readonly session: Session;
}
