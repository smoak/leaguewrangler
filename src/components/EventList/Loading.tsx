import { Box } from '@mui/material';
import SkeletonEvent from '../SkeletonEvent';

export const Loading = () => {
  return (
    <Box
      sx={{ display: 'grid', gap: '1.25rem', gridTemplateColumns: 'repeat(auto-fill,minmax(300px,1fr))' }}
      data-testid="skeleton-event-1"
    >
      <SkeletonEvent />
      <SkeletonEvent />
      <SkeletonEvent />
    </Box>
  );
};
