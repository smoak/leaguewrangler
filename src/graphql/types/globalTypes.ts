export type Maybe<T> = T | null;
export type InputMaybe<T> = Maybe<T>;
export type Exact<T extends { [key: string]: unknown }> = { [K in keyof T]: T[K] };
export type MakeOptional<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]?: Maybe<T[SubKey]> };
export type MakeMaybe<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]: Maybe<T[SubKey]> };
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
  ID: string;
  String: string;
  Boolean: boolean;
  Int: number;
  Float: number;
};

export type Query = {
  __typename?: 'Query';
  /** Gets the current user */
  currentUser: User;
  /** Get a single event */
  event?: Maybe<Event>;
  /** Get events for a team */
  events: Array<Maybe<Event>>;
  /** Get teams for a user */
  teams: Array<Maybe<Team>>;
};


export type QueryEventArgs = {
  eventId: Scalars['Int'];
  teamId: Scalars['Int'];
};


export type QueryEventsArgs = {
  teamId: Scalars['Int'];
};


export type QueryTeamsArgs = {
  dashboardOnly?: InputMaybe<Scalars['Boolean']>;
};

/** A user */
export type User = {
  __typename?: 'User';
  displayName: Scalars['String'];
  emailAddress1: Scalars['String'];
  events: Array<Maybe<Event>>;
  firstName: Scalars['String'];
  fullName: Scalars['String'];
  lastName: Scalars['String'];
  profilePhotoFullUrl: Scalars['String'];
  profilePhotoSmallUrl: Scalars['String'];
  profilePhotoThumbnailUrl: Scalars['String'];
  userId: Scalars['Int'];
};


/** A user */
export type UserEventsArgs = {
  dashboardOnly?: InputMaybe<Scalars['Boolean']>;
};

/** An event */
export type Event = {
  __typename?: 'Event';
  endTimestamp?: Maybe<Scalars['Int']>;
  eventId: Scalars['Int'];
  eventType: Scalars['String'];
  location?: Maybe<Location>;
  seasonId: Scalars['Int'];
  seasonName?: Maybe<Scalars['String']>;
  startTimestamp?: Maybe<Scalars['Int']>;
  status: Scalars['String'];
  team: Team;
  title: Scalars['String'];
  viewerRsvpStatus: RsvpStatus;
};

/** A location */
export type Location = {
  __typename?: 'Location';
  address?: Maybe<Scalars['String']>;
  googleMapsDirectionsUrl?: Maybe<Scalars['String']>;
  googleMapsUrl?: Maybe<Scalars['String']>;
  isPublic: Scalars['Boolean'];
  locationId: Scalars['Int'];
  name: Scalars['String'];
};

/** A team */
export type Team = {
  __typename?: 'Team';
  name: Scalars['String'];
  photoFullUrl?: Maybe<Scalars['String']>;
  photoSmallUrl?: Maybe<Scalars['String']>;
  photoThumbnailUrl?: Maybe<Scalars['String']>;
  shortName: Scalars['String'];
  teamId: Scalars['Int'];
};

/** The RSVP status of a user */
export enum RsvpStatus {
  /** Indicates the user is available to attend */
  AVAILABLE = 'AVAILABLE',
  /** Indicates the user might attend */
  MAYBE = 'MAYBE',
  /** Indicates the user will not attend */
  NO = 'NO',
  /** Indicates the user has not RSVPed */
  NONE = 'NONE',
  /** Indicates the user will attend */
  YES = 'YES'
}

export type Mutation = {
  __typename?: 'Mutation';
  /** Create a user token */
  createUserToken?: Maybe<UserToken>;
  /** RSVP to an event */
  saveRsvp: Event;
};


export type MutationCreateUserTokenArgs = {
  password: Scalars['String'];
  username: Scalars['String'];
};


export type MutationSaveRsvpArgs = {
  comments?: InputMaybe<Scalars['String']>;
  eventId: Scalars['Int'];
  rsvpStatus: RsvpStatus;
  teamId: Scalars['Int'];
};

/** A user token */
export type UserToken = {
  __typename?: 'UserToken';
  /** A unique token for the user matched. Tokens are 36-character, lower-cased GUIDs. */
  token: Scalars['String'];
};
