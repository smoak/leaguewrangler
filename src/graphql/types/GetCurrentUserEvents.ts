/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

import { RsvpStatus } from "./globalTypes";

// ====================================================
// GraphQL query operation: GetCurrentUserEvents
// ====================================================

export interface GetCurrentUserEvents_currentUser_events_team {
  __typename: "Team";
  shortName: string;
  photoThumbnailUrl: string | null;
}

export interface GetCurrentUserEvents_currentUser_events_location {
  __typename: "Location";
  name: string;
  address: string | null;
  googleMapsUrl: string | null;
}

export interface GetCurrentUserEvents_currentUser_events {
  __typename: "Event";
  eventId: number;
  title: string;
  endTimestamp: number | null;
  startTimestamp: number | null;
  viewerRsvpStatus: RsvpStatus;
  team: GetCurrentUserEvents_currentUser_events_team;
  location: GetCurrentUserEvents_currentUser_events_location | null;
}

export interface GetCurrentUserEvents_currentUser {
  __typename: "User";
  events: (GetCurrentUserEvents_currentUser_events | null)[];
}

export interface GetCurrentUserEvents {
  /**
   * Gets the current user
   */
  currentUser: GetCurrentUserEvents_currentUser;
}
