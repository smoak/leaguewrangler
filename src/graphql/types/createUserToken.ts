import * as Types from './globalTypes';

export type CreateUserTokenMutationVariables = Types.Exact<{
  username: Types.Scalars['String'];
  password: Types.Scalars['String'];
}>;


export type CreateUserTokenMutation = { __typename?: 'Mutation', createUserToken?: { __typename?: 'UserToken', token: string } | null };
