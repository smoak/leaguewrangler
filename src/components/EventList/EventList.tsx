import React, { FC } from 'react';
import { useQuery } from '@apollo/react-hooks';
import { List, ListItem } from '@material-ui/core';

import { GetCurrentUserEvents } from '../../graphql/types/GetCurrentUserEvents';
import getCurrentUserEvents from '../../graphql/queries/getCurrentUserEvents';
import Event from '../Event';
import SkeletonEvent from '../SkeletonEvent';

function notEmpty<TValue>(value: TValue | null | undefined): value is TValue {
  return value !== null && value !== undefined;
}

const EventList: FC = () => {
  const { loading, data, error } = useQuery<GetCurrentUserEvents>(getCurrentUserEvents);

  if (loading) {
    return <SkeletonEvent />;
  }

  if (error || !data) {
    return <>Some error</>;
  }

  const { currentUser } = data;

  const events = currentUser.events.filter(notEmpty);

  if (events.length > 0) {
    return (
      <List>
        {events.map(e => (
          <ListItem key={e.eventId}>
            <Event
              title={e.title}
              teamPhotoUrl={e.team.photoThumbnailUrl}
              location={e.location ? e.location.name : ''}
              startTime={e.startTimestamp ? new Date(e.startTimestamp * 1000) : undefined}
              rsvpStatus={e.viewerRsvpStatus}
            />
          </ListItem>
        ))}
      </List>
    );
  }

  return <>No events scheduled</>;
};

export default EventList;
