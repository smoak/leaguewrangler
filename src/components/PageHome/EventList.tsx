import React, { FC } from 'react';
import getCurrentUserEvents from '../../graphql/queries/getCurrentUserEvents';
import { ChildDataProps, withQuery } from 'react-apollo';
import { GetCurrentUserEvents } from '../../graphql/types/GetCurrentUserEvents';
import { CircularProgress, List, ListItem } from '@material-ui/core';
import Event from '../Event';

export type EventProps = ChildDataProps<{}, GetCurrentUserEvents, {}>;

function notEmpty<TValue>(value: TValue | null | undefined): value is TValue {
  return value !== null && value !== undefined;
}

const EventList: FC<EventProps> = ({ data: { loading, error, currentUser } }) => {
  if (loading) {
    return <CircularProgress color="secondary" />;
  }

  if (error || !currentUser) {
    return <>'Some error'</>;
  }

  const events = currentUser.events.filter(notEmpty);

  if (events.length > 0) {
    return (
      <List>
        {events.map(e => (
          <ListItem>
            <Event
              title={e.title.replace('&nbsp;', ' ')}
              teamPhotoUrl={e.team.photoThumbnailUrl}
              location={e.location ? e.location.name : ''}
              startTime={e.startTimestamp ? new Date(e.startTimestamp * 1000) : undefined}
            />
          </ListItem>
        ))}
      </List>
    );
  }

  return <>No events scheduled</>;
};

export default withQuery<{}, GetCurrentUserEvents, {}>(getCurrentUserEvents)(EventList);
