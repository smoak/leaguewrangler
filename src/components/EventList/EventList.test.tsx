import { useQuery } from '@apollo/react-hooks';
import { ShallowWrapper, shallow } from 'enzyme';

import { GetCurrentUserEvents } from '../../graphql/types/GetCurrentUserEvents';
import { RsvpStatus } from '../../graphql/types/globalTypes';

import EventList from './EventList';

describe('EventList', () => {
  let component: ShallowWrapper;

  describe('when the data is loading', () => {
    beforeEach(() => {
      (useQuery as jest.Mock).mockReturnValue({ loading: true });

      component = shallow(<EventList />);
    });

    it('renders as expected', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('when the server returns an error', () => {
    beforeEach(() => {
      (useQuery as jest.Mock).mockReturnValue({ loading: false, error: 'nope' });

      component = shallow(<EventList />);
    });

    it('renders as expected', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('when the server returns no data', () => {
    beforeEach(() => {
      (useQuery as jest.Mock).mockReturnValue({ loading: false, error: undefined, data: null });

      component = shallow(<EventList />);
    });

    it('renders as expected', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('when the server returns data', () => {
    describe('and the user has no events', () => {
      beforeEach(() => {
        const data: GetCurrentUserEvents = {
          currentUser: {
            events: [],
            __typename: 'User',
          },
        };
        (useQuery as jest.Mock).mockReturnValue({ loading: false, error: undefined, data });

        component = shallow(<EventList />);
      });

      it('renders as expected', () => {
        expect(component).toMatchSnapshot();
      });
    });

    describe('and the user has events', () => {
      beforeEach(() => {
        const data: GetCurrentUserEvents = {
          currentUser: {
            events: [
              {
                __typename: 'Event',
                endTimestamp: 1657691037616,
                startTimestamp: 1657691025566,
                id: 1,
                location: null,
                team: {
                  __typename: 'Team',
                  id: 1,
                  photoThumbnailUrl: 'url',
                  shortName: 'team',
                },
                title: 'test',
                viewerRsvpStatus: RsvpStatus.NONE,
              },
            ],
            __typename: 'User',
          },
        };
        (useQuery as jest.Mock).mockReturnValue({ loading: false, error: undefined, data });

        component = shallow(<EventList />);
      });

      it('renders as expected', () => {
        expect(component).toMatchSnapshot();
      });
    });

    describe('when the event has no start time', () => {
      beforeEach(() => {
        const data: GetCurrentUserEvents = {
          currentUser: {
            events: [
              {
                __typename: 'Event',
                endTimestamp: 1657691037616,
                startTimestamp: null,
                id: 1,
                location: {
                  __typename: 'Location',
                  name: 'test location',
                  address: '123 test',
                  googleMapsUrl: 'url',
                },
                team: {
                  __typename: 'Team',
                  id: 1,
                  photoThumbnailUrl: 'url',
                  shortName: 'team',
                },
                title: 'test',
                viewerRsvpStatus: RsvpStatus.NONE,
              },
            ],
            __typename: 'User',
          },
        };
        (useQuery as jest.Mock).mockReturnValue({ loading: false, error: undefined, data });

        component = shallow(<EventList />);
      });

      it('renders as expected', () => {
        expect(component).toMatchSnapshot();
      });
    });

    describe('when the event has a location', () => {
      beforeEach(() => {
        const data: GetCurrentUserEvents = {
          currentUser: {
            events: [
              {
                __typename: 'Event',
                endTimestamp: 1657691037616,
                startTimestamp: 1657691025566,
                id: 1,
                location: {
                  __typename: 'Location',
                  name: 'test location',
                  address: '123 test',
                  googleMapsUrl: 'url',
                },
                team: {
                  __typename: 'Team',
                  id: 1,
                  photoThumbnailUrl: 'url',
                  shortName: 'team',
                },
                title: 'test',
                viewerRsvpStatus: RsvpStatus.NONE,
              },
            ],
            __typename: 'User',
          },
        };
        (useQuery as jest.Mock).mockReturnValue({ loading: false, error: undefined, data });

        component = shallow(<EventList />);
      });

      it('renders as expected', () => {
        expect(component).toMatchSnapshot();
      });
    });

    it('renders as expected', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
