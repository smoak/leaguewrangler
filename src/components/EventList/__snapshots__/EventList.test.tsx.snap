// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventList when the data is loading renders as expected 1`] = `<WithStyles(SkeletonEvent) />`;

exports[`EventList when the server returns an error renders as expected 1`] = `
<Fragment>
  Some error
</Fragment>
`;

exports[`EventList when the server returns data and the user has events renders as expected 1`] = `
<WithStyles(ForwardRef(List))>
  <WithStyles(ForwardRef(ListItem))
    key="1"
  >
    <WithStyles(Event)
      eventId={1}
      location=""
      rsvpStatus="NONE"
      startTime={1970-01-01T00:00:02.000Z}
      teamId={1}
      teamPhotoUrl="url"
      title="test"
    />
  </WithStyles(ForwardRef(ListItem))>
</WithStyles(ForwardRef(List))>
`;

exports[`EventList when the server returns data and the user has no events renders as expected 1`] = `
<Fragment>
  No events scheduled
</Fragment>
`;

exports[`EventList when the server returns data renders as expected 1`] = `
<WithStyles(ForwardRef(List))>
  <WithStyles(ForwardRef(ListItem))
    key="1"
  >
    <WithStyles(Event)
      eventId={1}
      location="test location"
      locationMapsUrl="url"
      rsvpStatus="NONE"
      startTime={1970-01-01T00:00:02.000Z}
      teamId={1}
      teamPhotoUrl="url"
      title="test"
    />
  </WithStyles(ForwardRef(ListItem))>
</WithStyles(ForwardRef(List))>
`;

exports[`EventList when the server returns data when the event has a location renders as expected 1`] = `
<WithStyles(ForwardRef(List))>
  <WithStyles(ForwardRef(ListItem))
    key="1"
  >
    <WithStyles(Event)
      eventId={1}
      location="test location"
      locationMapsUrl="url"
      rsvpStatus="NONE"
      startTime={1970-01-01T00:00:02.000Z}
      teamId={1}
      teamPhotoUrl="url"
      title="test"
    />
  </WithStyles(ForwardRef(ListItem))>
</WithStyles(ForwardRef(List))>
`;

exports[`EventList when the server returns data when the event has no start time renders as expected 1`] = `
<WithStyles(ForwardRef(List))>
  <WithStyles(ForwardRef(ListItem))
    key="1"
  >
    <WithStyles(Event)
      eventId={1}
      location="test location"
      locationMapsUrl="url"
      rsvpStatus="NONE"
      teamId={1}
      teamPhotoUrl="url"
      title="test"
    />
  </WithStyles(ForwardRef(ListItem))>
</WithStyles(ForwardRef(List))>
`;

exports[`EventList when the server returns no data renders as expected 1`] = `
<Fragment>
  Some error
</Fragment>
`;
